testing/tests/DevExpress.ui.widgets.form/form.API.option_formData.tests.js

```diff
@@ -329,3 +329,139 @@ QUnit.test('Reset editor\'s value when set formData: {dataField1: a}', function(
+    }
+
+    function createTestData(boolValue) {
+        return boolValue !== 'no member'
+            ? { b: boolValue }
+            : {};
+    }
+
+    function getItemByType(itemType, item) {
+        switch(itemType) {
+            case 'simpleItem':
+                return item;
+            case 'groupItem':
+                return { itemType: 'group', items: [ item ] };
+            case 'tabbedItem':
+                return { itemType: 'tabbed', tabs: [{ items: [ item ] }] };
+        }
+    }
+
+    [true, false, undefined].forEach(allowIndeterminateState => {
+        ['simpleItem', 'groupItem', 'tabbedItem'].forEach(itemType => {
+            [true, false, undefined, null, 'no member'].forEach(oldBoolValue => {
+                [true, false, undefined, null, 'no member'].forEach(newBoolValue => {
+                    QUnit.test(`AllowIndeterminateState = ${allowIndeterminateState}, ItemType: ${itemType}, FormData = { b:  ${oldBoolValue}} -> updateFormData({ b: ${newBoolValue} })`, function(assert) {
+                        const form = $('#form').dxForm({
+                            formData: createTestData(oldBoolValue),
+                            items: [ getItemByType(itemType,
+                                { dataField: 'b', editorType: 'dxCheckBox', allowIndeterminateState: allowIndeterminateState })
+                            ]
+                        }).dxForm('instance');
+
+                        const editorOldValue = (oldBoolValue === 'no member' || (allowIndeterminateState === false && oldBoolValue === undefined))
+                            ? false
+                            : oldBoolValue;
+                        checkEditorAndFormValue(form, 'b', editorOldValue, oldBoolValue);
+
+                        const newFormData = {};
+                        if(newBoolValue !== 'no member') {
+                            newFormData['b'] = newBoolValue;
+                        }
+
+                        form.updateData(newFormData);
+                        const editorNewValue = newBoolValue === 'no member' ? editorOldValue : newBoolValue;
+                        const formNewValue = newBoolValue === 'no member' ? oldBoolValue : newBoolValue;
+                        checkEditorAndFormValue(form, 'b', editorNewValue, formNewValue);
+                    });
+
+                    QUnit.test(`AllowIndeterminateState = ${allowIndeterminateState}, ItemType: ${itemType}, FormData = { innerObject: { b:  ${oldBoolValue} }} -> updateFormData({ innerObject.b = ${newBoolValue })`, function(assert) {
+                        const form = $('#form').dxForm({
+                            formData: { innerObject: createTestData(oldBoolValue) },
+                            items: [ getItemByType(itemType,
+                                { dataField: 'innerObject.b', editorType: 'dxCheckBox', allowIndeterminateState: allowIndeterminateState })
```

Comment:
Imho we don't need to test 'innerObject.member' configs and it's enough to test 'member' configs only at the current stage.